AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Gamemode_Timing_Test_Template.py

```diff
@@ -0,0 +1,133 @@
+    """
+    Summary:
+    Loads the "Target_Level" defined above within an instance of Editor.exe using the null renderer. Test verifies that
+    the level loads, can enter/exit gameplay, and that Editor.exe remains stable throughout this process.
+
+    Test setup:
+    - Launch Editor.exe
+
+    Expected Behavior:
+    Test verifies that level loads, enters/exits game mode, and editor remains stable.
+
+    Test Steps:
+    1) Load level, confirm that correct level is loaded, and report results
+    2) Start timer for enter game mode.
+    3) Validate that editor can enter gameplay successfully
+    4) Checks if the timeout has been reached (default 10 sec), if not report time editor took to enter game mode.
+    5) Start timer for exit game mode
+    6) Validate that editor can exit gameplay successfully
+    7) Checks if the timeout has been reached (default 10 sec), if not report time editor took to exit game mode.
+    8) Open an empty level to verify editor remains stable
+    9) Look for errors or asserts.
+
+    :return: None
+    """
+
+    import azlmbr.legacy.general as general
+    import time
+
+    from editor_python_test_tools.utils import Report, Tracer, TestHelper
+
+    # Timer class to track enter/exit game time
+    class Timer:
+        unit_divisor = 60
+        hour_divisor = unit_divisor * unit_divisor
+
+        def start(self):
+            self._start_time = time.perf_counter()
+
+        def stop(self):
+            self._stop_time = time.perf_counter() - self._start_time
+            return self._stop_time
+
+        def log_time(self, message):
+            elapsed_time = time.perf_counter() - self._start_time
+            hours = int(elapsed_time / Timer.hour_divisor)
+            minutes = int(elapsed_time % Timer.hour_divisor / Timer.unit_divisor)
+            seconds = elapsed_time % Timer.unit_divisor
+
+            return(f'{message}: {hours:0>2d}:{minutes:0>2d}:{seconds:0>5.2f}\n')
+
+        def just_seconds(self):
+            elapsed_time = time.perf_counter() - self._start_time
+            seconds = elapsed_time % Timer.unit_divisor
+
+            return seconds
+
+    with Tracer() as error_tracer:
+
+        # 1. Load target level, confirm that correct level is loaded, and report results.
+        TestHelper.init_idle()
+        TestHelper.open_level("Graphics", Target_Level)
+        Report.result(Tests.target_level_load, Target_Level
+                      == general.get_current_level_name())
+
+        # 2. Start timer for enter game mode.
+        enter_gamemode_length = Timer()
+        enter_gamemode_length.start()
+
+        # 3. Validate that editor can enter gameplay successfully.
+        TestHelper.enter_game_mode(Tests.target_level_enter_game_mode)
+
+        # 4. Checks if the timeout has been reached (default 10 sec), if not report time editor took to enter game mode.
+        Report.critical_result(Tests.timeout_status, enter_gamemode_length.stop() < 10.0)
```

Comment:
if TestHelper.enter_game_mode includes a wait_for_condition that's stopping you from timing on the is in game mode condition then maybe directly use the azlmbr.legacy.general.enter_game_mode() with a wait_for_condition.
if TestHelper.enter_game_mode is failing a critical_result on 1 frame delay like it looks like it does.
the critical_result here means the test will stop if we exceed the 10 second metric; no data will be reported

Reply:
I don't understand why this is a problem, and why wait for condition is needed? seems to needlessly complicate things

Also if it exceeded the timeout, do we care by how much?

If that is the case why have a timeout at all?

Reply:
I thought you had to wait for general.is_in_game_mode() to be true, but if general.enter_game_mode() is synchronous and doesn't return until you are in game mode I guess you can avoid having to wait on a condition.
by using a critical_result here if the enter game mode is >= 10.0 you abort any other testing like the exit game mode. the point of a critial_result would be if you had a wait_for_condition with a timeout of 30 seconds it would fail only if the enter game mode was horribly long where the target is 10 seconds.
Performance metrics are goals. you still want to track how far away you are as this may not always pass the 10 second bar